In [42]:
import pandas as pd
import numpy as np

In [3]:
df_sea_ice = pd.read_csv("sea_ice_extent.csv")

In [4]:
df_sea_ice.head()

Year   Month   Day       Extent      Missing  \
0  1978      10    26       10.231          0.0   
1  1978      10    28       10.420          0.0   
2  1978      10    30       10.557          0.0   
3  1978      11     1       10.670          0.0   
4  1978      11     3       10.777          0.0   

                                         Source Data hemisphere  
0   ['ftp://sidads.colorado.edu/pub/DATASETS/nsid...      north  
1   ['ftp://sidads.colorado.edu/pub/DATASETS/nsid...      north  
2   ['ftp://sidads.colorado.edu/pub/DATASETS/nsid...      north  
3   ['ftp://sidads.colorado.edu/pub/DATASETS/nsid...      north  
4   ['ftp://sidads.colorado.edu/pub/DATASETS/nsid...      north

In [5]:
df_sea_ice.columns = df_sea_ice.columns.str.strip()  # Boşlukları temizle


In [6]:
print(df_sea_ice.columns)

Index(['Year', 'Month', 'Day', 'Extent', 'Missing', 'Source Data',
       'hemisphere'],
      dtype='object')


In [7]:
df_sea_ice.drop(['Missing', 'Source Data'], axis=1, inplace=True)
df_sea_ice.head()

Year  Month  Day  Extent hemisphere
0  1978     10   26  10.231      north
1  1978     10   28  10.420      north
2  1978     10   30  10.557      north
3  1978     11    1  10.670      north
4  1978     11    3  10.777      north

In [8]:
df_sea_ice = df_sea_ice.groupby(['Year', 'Month', 'hemisphere'])['Extent'].mean().reset_index()
print(df_sea_ice)


     Year  Month hemisphere     Extent
0    1978     10      north  10.402667
1    1978     10      south  17.699000
2    1978     11      north  11.645133
3    1978     11      south  15.896400
4    1978     12      north  13.667063
..    ...    ...        ...        ...
971  2019      3      south   3.164161
972  2019      4      north  13.454600
973  2019      4      south   5.708633
974  2019      5      north  12.163161
975  2019      5      south   8.835516

[976 rows x 4 columns]


### df_sea_ice ın son hali

In [9]:
df_sea_ice = df_sea_ice.groupby(['Year', 'hemisphere'])['Extent'].mean().reset_index()
print(df_sea_ice)

    Year hemisphere     Extent
0   1978      north  11.904954
1   1978      south  14.663550
2   1979      north  12.349634
3   1979      south  11.653819
4   1980      north  12.349545
..   ...        ...        ...
79  2017      south  10.701843
80  2018      north  10.349772
81  2018      south  10.889444
82  2019      north  13.624753
83  2019      south   4.839170

[84 rows x 3 columns]


#### doğru mu yaptık
- 1978/10 north 10.402667
- 1978/11 north 11.645133
- 1978/12 north 13.667063
- 13.667063+11.645133+10.402667=35714853
- 35714863/3=11904954


_____________________

In [10]:
df_northern_land = pd.read_csv("data_northern_land_temp.csv")


In [11]:
df_northern_land.head()

Year  Anomaly
0  197801     0.10
1  197802     0.07
2  197803     0.37
3  197804     0.21
4  197805    -0.04

year ve ay birleşik yazılmış. datetime kullanmamız lazım

In [12]:
df_northern_land['Year'] = pd.to_datetime(df_northern_land['Year'], format='%Y%m')
df_northern_land.head()


Year  Anomaly
0 1978-01-01     0.10
1 1978-02-01     0.07
2 1978-03-01     0.37
3 1978-04-01     0.21
4 1978-05-01    -0.04

In [13]:
df_northern_land = df_northern_land.groupby(df_northern_land['Year'].dt.year)['Anomaly'].mean().reset_index()
df_northern_land.head()

Year   Anomaly
0  1978  0.010833
1  1979  0.042500
2  1980  0.259167
3  1981  0.714167
4  1982 -0.011667

#### doğru mu
- 1978 deki tüm ayların toplamı:0.12
- 0.13/12=0.010833
- yani doğru

### df_northern_land in son hali

In [14]:
df_northern_land['hemisphere'] = 'north'
df_northern_land.rename(columns={'Anomaly': 'AnomalyLand'}, inplace=True)

df_northern_land.head()

Year  AnomalyLand hemisphere
0  1978     0.010833      north
1  1979     0.042500      north
2  1980     0.259167      north
3  1981     0.714167      north
4  1982    -0.011667      north

In [15]:
df_northern_ocean = pd.read_csv("data_northern_ocean_temp.csv")
df_northern_ocean['Year'] = pd.to_datetime(df_northern_ocean['Year'], format='%Y%m')
df_northern_ocean = df_northern_ocean.groupby(df_northern_ocean['Year'].dt.year)['Anomaly'].mean().reset_index()



#### df_northern_ocean ın son hali

In [16]:
df_northern_ocean['hemisphere'] = 'north'
df_northern_ocean.rename(columns={'Anomaly': 'AnomalyOcean'}, inplace=True)

df_northern_ocean.head()

Year  AnomalyOcean hemisphere
0  1978      0.085000      north
1  1979      0.188333      north
2  1980      0.182500      north
3  1981      0.197500      north
4  1982      0.165000      north

#### df_southern_land in son hali

In [17]:
df_southern_land = pd.read_csv("data_southern_land_temp.csv")
df_southern_land['Year'] = pd.to_datetime(df_southern_land['Year'], format='%Y%m')
df_southern_land = df_southern_land.groupby(df_southern_land['Year'].dt.year)['Anomaly'].mean().reset_index()
df_southern_land['hemisphere'] = 'south'
df_southern_land.rename(columns={'Anomaly': 'AnomalyLand'}, inplace=True)

df_southern_land.head()


Year  AnomalyLand hemisphere
0  1978     0.183333      south
1  1979     0.299167      south
2  1980     0.426667      south
3  1981     0.264167      south
4  1982     0.270000      south

#### df_southern_ocean ın son hali

In [18]:
df_southern_ocean = pd.read_csv("data_southern_ocean_temp.csv")
df_southern_ocean['Year'] = pd.to_datetime(df_southern_ocean['Year'], format='%Y%m')
df_southern_ocean = df_southern_ocean.groupby(df_southern_ocean['Year'].dt.year)['Anomaly'].mean().reset_index()
df_southern_ocean['hemisphere'] = 'south'
df_southern_ocean.rename(columns={'Anomaly': 'AnomalyOcean'}, inplace=True)

df_southern_ocean.head()


Year  AnomalyOcean hemisphere
0  1978      0.175000      south
1  1979      0.330000      south
2  1980      0.357500      south
3  1981      0.293333      south
4  1982      0.294167      south

In [19]:
df_land_temp = pd.concat([df_northern_land, df_southern_land], ignore_index=True)
df_land_temp.head()

Year  AnomalyLand hemisphere
0  1978     0.010833      north
1  1979     0.042500      north
2  1980     0.259167      north
3  1981     0.714167      north
4  1982    -0.011667      north

In [20]:
df_ocean_temp = pd.concat([df_northern_ocean, df_southern_ocean], ignore_index=True)
df_ocean_temp.head()

Year  AnomalyOcean hemisphere
0  1978      0.085000      north
1  1979      0.188333      north
2  1980      0.182500      north
3  1981      0.197500      north
4  1982      0.165000      north

In [21]:
df_annual_afforest = pd.read_csv("annual-afforestation.csv")

In [22]:
df_annual_afforest.head()

Entity Code  Year  Afforestation
0  Afghanistan  AFG  2000             70
1  Afghanistan  AFG  2010             70
2  Afghanistan  AFG  2015             70
3      Albania  ALB  1990            800
4      Albania  ALB  2000            800

In [29]:
df_annual_afforest["Entity"].value_counts()

Maldives      4
Morocco       4
Norway        4
Niger         4
Nicaragua     4
             ..
Guinea        1
Belgium       1
Madagascar    1
Mozambique    1
Portugal      1
Name: Entity, Length: 105, dtype: int64

In [31]:
df_annual_afforest["Year"].value_counts()

2010    98
2000    97
2015    90
1990    81
Name: Year, dtype: int64

In [32]:
df_annual_deforest = pd.read_csv("annual-deforestation.csv")


In [33]:
df_annual_deforest.head()

Entity Code  Year  Deforestation
0   Africa  NaN  1990        4096000
1   Africa  NaN  2000        4314000
2   Africa  NaN  2010        4444000
3   Africa  NaN  2015        4414000
4  Algeria  DZA  1990          14800

In [34]:
df_annual_deforest["Year"].value_counts()

2010    133
2000    127
2015    126
1990    109
Name: Year, dtype: int64

## bu annual afforestation ve deforestation verisetlerini kullanmayacağız. yılları bizim setimizi çok kapsamıyor.

In [21]:
df_fosil_s = pd.read_csv("Total_Fossil_Industy.csv")

In [22]:
df_fosil_s.head()

Year  Fossil and Industrial
0  1750               9.505620
1  1751               9.505963
2  1752               9.510002
3  1753               9.510345
4  1754               9.514384

In [23]:
df_fosil_s["hemisphere"] = 'south'
df_fosil_s.head()

Year  Fossil and Industrial hemisphere
0  1750               9.505620      south
1  1751               9.505963      south
2  1752               9.510002      south
3  1753               9.510345      south
4  1754               9.514384      south

In [24]:
df_fosil_n = pd.read_csv("Total_Fossil_Industy.csv")


In [25]:
df_fosil_n["hemisphere"] = 'north'
df_fosil_n.head()

Year  Fossil and Industrial hemisphere
0  1750               9.505620      north
1  1751               9.505963      north
2  1752               9.510002      north
3  1753               9.510345      north
4  1754               9.514384      north

In [26]:
df_fosil = pd.concat([df_fosil_n, df_fosil_s], ignore_index=True)

### df_fosil son hali

In [27]:
df_fosil.tail()

Year  Fossil and Industrial hemisphere
537  2016             35702.1833      south
538  2017             35807.3679      south
539  2018             36456.9087      south
540  2019             36908.1024      south
541  2020             37388.1289      south

### df_agri son hali

In [28]:
df_agri_s = pd.read_csv('Total_Agriculture_forest.csv')
df_agri_s["hemisphere"] = 'south'
df_agri_n = pd.read_csv('Total_Agriculture_forest.csv')
df_agri_n["hemisphere"] = 'north'
df_agri = pd.concat([df_agri_n, df_agri_s], ignore_index=True)
df_agri.head()

Year  Agriculture Waste and Forest Burining hemisphere
0  1750                             297.464607      north
1  1751                             302.937098      north
2  1752                             308.510269      north
3  1753                             314.185969      north
4  1754                             319.966086      north

### concat all

In [29]:
merged_df = pd.merge(df_sea_ice, df_land_temp,df_ocean_temp, df_fosil, df_agri, on=['Year', 'hemisphere'], how='inner')


TypeError: merge() got multiple values for argument 'on'

In [30]:
merged_df = pd.merge(df_sea_ice, df_land_temp, on=['Year', 'hemisphere'], how='inner')
merged_df = pd.merge(merged_df, df_ocean_temp, on=['Year', 'hemisphere'], how='inner')
merged_df = pd.merge(merged_df, df_fosil, on=['Year', 'hemisphere'], how='inner')
merged_df = pd.merge(merged_df, df_agri, on=['Year', 'hemisphere'], how='inner')


In [31]:
merged_df.head()

Year hemisphere     Extent  AnomalyLand  AnomalyOcean  \
0  1978      north  11.904954     0.010833      0.085000   
1  1978      south  14.663550     0.183333      0.175000   
2  1979      north  12.349634     0.042500      0.188333   
3  1979      south  11.653819     0.299167      0.330000   
4  1980      north  12.349545     0.259167      0.182500   

   Fossil and Industrial  Agriculture Waste and Forest Burining  
0            20454.21696                            4761.222033  
1            20454.21696                            4761.222033  
2            20999.96016                            4589.050767  
3            20999.96016                            4589.050767  
4            20707.59534                            4559.068800

In [43]:
df_co= pd.read_csv("CO2_global_CEDS_emissions_by_fuel_2021_04_21.csv")

In [44]:
df_co.head()

fuel   em  units        X1750        X1751        X1752        X1753  \
0     biomass  CO2  ktCO2     0.000000     0.000000     0.000000     0.000000   
1  brown_coal  CO2  ktCO2     0.000000     0.000000     0.000000     0.000000   
2   coal_coke  CO2  ktCO2   366.177444   366.177444   366.320930   366.320930   
3  diesel_oil  CO2  ktCO2     0.000000     0.000000     0.000000     0.000000   
4   hard_coal  CO2  ktCO2  6735.455519  6735.455519  6738.094804  6738.094804   

         X1754        X1755        X1756  ...         X2010         X2011  \
0     0.000000     0.000000     0.000000  ...  0.000000e+00  0.000000e+00   
1     0.000000     0.000000     0.000000  ...  7.473192e+05  7.867700e+05   
2   366.464417   366.607903   391.861520  ...  4.246643e+05  4.707760e+05   
3     0.000000     0.000000     0.000000  ...  3.876727e+06  3.971659e+06   
4  6740.734089  6743.373374  7207.887548  ...  1.057572e+07  1.105209e+07   

          X2012         X2013         X2014         X2015         X2016  \
0  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00   
1  7.787084e+05  7.362840e+05  7.196863e+05  7.207185e+05  7.154591e+05   
2  5.624562e+05  4.937008e+05  5.116548e+05  5.310231e+05  5.469573e+05   
3  3.983097e+06  4.096506e+06  4.125405e+06  4.231086e+06  4.197037e+06   
4  1.112120e+07  1.145373e+07  1.149486e+07  1.123129e+07  1.092686e+07   

          X2017         X2018         X2019  
0  0.000000e+00  0.000000e+00  0.000000e+00  
1  7.182963e+05  7.410650e+05  6.987816e+05  
2  5.434647e+05  5.489173e+05  5.533338e+05  
3  4.279408e+06  4.339067e+06  4.372833e+06  
4  1.103649e+07  1.122733e+07  1.118936e+07  

[5 rows x 273 columns]

In [33]:
df_melted = df_co.melt(id_vars=['fuel', 'em', 'units'], var_name='Year', value_name='Value')

# year sıralayalım
df_melted['Year'] = df_melted['Year'].str.extract('X(\d+)', expand=False).astype(int)

#transpozunu alacağım
df_transposed = df_melted.pivot(index='Year', columns='fuel', values='Value')

# year tekrar eklensin
df_co_res = df_transposed.reset_index().rename_axis(columns=None)

print(df_co_res)

     Year  biomass     brown_coal      coal_coke    diesel_oil     hard_coal  \
0    1750      0.0       0.000000     366.177444  0.000000e+00  6.735456e+03   
1    1751      0.0       0.000000     366.177444  0.000000e+00  6.735456e+03   
2    1752      0.0       0.000000     366.320930  0.000000e+00  6.738095e+03   
3    1753      0.0       0.000000     366.320930  0.000000e+00  6.738095e+03   
4    1754      0.0       0.000000     366.464417  0.000000e+00  6.740734e+03   
..    ...      ...            ...            ...           ...           ...   
265  2015      0.0  720718.475341  531023.090905  4.231086e+06  1.123129e+07   
266  2016      0.0  715459.119445  546957.337511  4.197037e+06  1.092686e+07   
267  2017      0.0  718296.340601  543464.732025  4.279408e+06  1.103649e+07   
268  2018      0.0  741064.980372  548917.287608  4.339067e+06  1.122733e+07   
269  2019      0.0  698781.625261  553333.808495  4.372833e+06  1.118936e+07   

        heavy_oil     light_oil   natur

In [34]:
df_melted = df_co.melt(id_vars=['fuel', 'em', 'units'], var_name='Year', value_name='Value')

# year sıralayalım
df_melted['Year'] = df_melted['Year'].str.extract('X(\d+)', expand=False).astype(int)

#transpozunu alacağım
df_transposed = df_melted.pivot(index='Year', columns='fuel', values='Value')

# year tekrar eklensin
df_co_res_s = df_transposed.reset_index().rename_axis(columns=None)
df_co_res_s['hemisphere'] ='south'

print(df_co_res_s)

     Year  biomass     brown_coal      coal_coke    diesel_oil     hard_coal  \
0    1750      0.0       0.000000     366.177444  0.000000e+00  6.735456e+03   
1    1751      0.0       0.000000     366.177444  0.000000e+00  6.735456e+03   
2    1752      0.0       0.000000     366.320930  0.000000e+00  6.738095e+03   
3    1753      0.0       0.000000     366.320930  0.000000e+00  6.738095e+03   
4    1754      0.0       0.000000     366.464417  0.000000e+00  6.740734e+03   
..    ...      ...            ...            ...           ...           ...   
265  2015      0.0  720718.475341  531023.090905  4.231086e+06  1.123129e+07   
266  2016      0.0  715459.119445  546957.337511  4.197037e+06  1.092686e+07   
267  2017      0.0  718296.340601  543464.732025  4.279408e+06  1.103649e+07   
268  2018      0.0  741064.980372  548917.287608  4.339067e+06  1.122733e+07   
269  2019      0.0  698781.625261  553333.808495  4.372833e+06  1.118936e+07   

        heavy_oil     light_oil   natur

In [35]:
df_melted = df_co.melt(id_vars=['fuel', 'em', 'units'], var_name='Year', value_name='Value')

# year sıralayalım
df_melted['Year'] = df_melted['Year'].str.extract('X(\d+)', expand=False).astype(int)

#transpozunu alacağım
df_transposed = df_melted.pivot(index='Year', columns='fuel', values='Value')

# year tekrar eklensin
df_co_res_n = df_transposed.reset_index().rename_axis(columns=None)
df_co_res_n['hemisphere'] ='north'

print(df_co_res_n)

     Year  biomass     brown_coal      coal_coke    diesel_oil     hard_coal  \
0    1750      0.0       0.000000     366.177444  0.000000e+00  6.735456e+03   
1    1751      0.0       0.000000     366.177444  0.000000e+00  6.735456e+03   
2    1752      0.0       0.000000     366.320930  0.000000e+00  6.738095e+03   
3    1753      0.0       0.000000     366.320930  0.000000e+00  6.738095e+03   
4    1754      0.0       0.000000     366.464417  0.000000e+00  6.740734e+03   
..    ...      ...            ...            ...           ...           ...   
265  2015      0.0  720718.475341  531023.090905  4.231086e+06  1.123129e+07   
266  2016      0.0  715459.119445  546957.337511  4.197037e+06  1.092686e+07   
267  2017      0.0  718296.340601  543464.732025  4.279408e+06  1.103649e+07   
268  2018      0.0  741064.980372  548917.287608  4.339067e+06  1.122733e+07   
269  2019      0.0  698781.625261  553333.808495  4.372833e+06  1.118936e+07   

        heavy_oil     light_oil   natur

In [36]:
df_co_all = pd.concat([df_co_res_n, df_co_res_s], ignore_index=True)

In [37]:
df_co_all.head()

Year  biomass  brown_coal   coal_coke  diesel_oil    hard_coal  heavy_oil  \
0  1750      0.0         0.0  366.177444         0.0  6735.455519        0.0   
1  1751      0.0         0.0  366.177444         0.0  6735.455519        0.0   
2  1752      0.0         0.0  366.320930         0.0  6738.094804        0.0   
3  1753      0.0         0.0  366.320930         0.0  6738.094804        0.0   
4  1754      0.0         0.0  366.464417         0.0  6740.734089        0.0   

   light_oil  natural_gas      process hemisphere  
0        0.0          0.0  1475.123768      north  
1        0.0          0.0  1475.123768      north  
2        0.0          0.0  1475.701795      north  
3        0.0          0.0  1475.701795      north  
4        0.0          0.0  1476.279821      north

In [38]:
merged_df = pd.merge(merged_df, df_co_all, on=['Year', 'hemisphere'], how='inner')

In [39]:
merged_df.head()

Year hemisphere     Extent  AnomalyLand  AnomalyOcean  \
0  1978      north  11.904954     0.010833      0.085000   
1  1978      south  14.663550     0.183333      0.175000   
2  1979      north  12.349634     0.042500      0.188333   
3  1979      south  11.653819     0.299167      0.330000   
4  1980      north  12.349545     0.259167      0.182500   

   Fossil and Industrial  Agriculture Waste and Forest Burining  biomass  \
0            20454.21696                            4761.222033      0.0   
1            20454.21696                            4761.222033      0.0   
2            20999.96016                            4589.050767      0.0   
3            20999.96016                            4589.050767      0.0   
4            20707.59534                            4559.068800      0.0   

      brown_coal      coal_coke    diesel_oil     hard_coal     heavy_oil  \
0  659352.812692  288821.546116  2.221299e+06  4.464100e+06  2.995678e+06   
1  659352.812692  288821.546116  2.221299e+06  4.464100e+06  2.995678e+06   
2  684733.458916  307656.730292  2.251136e+06  4.651720e+06  3.019362e+06   
3  684733.458916  307656.730292  2.251136e+06  4.651720e+06  3.019362e+06   
4  712064.633607  328489.122354  2.156050e+06  4.795404e+06  2.822541e+06   

      light_oil   natural_gas       process  
0  2.843968e+06  2.325158e+06  3.950627e+06  
1  2.843968e+06  2.325158e+06  3.950627e+06  
2  2.883021e+06  2.462239e+06  3.998047e+06  
3  2.883021e+06  2.462239e+06  3.998047e+06  
4  2.828348e+06  2.543048e+06  3.865657e+06

In [40]:
merged_df.to_csv('project1.csv',index=False)

In [46]:
merged_df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 84 entries, 0 to 83
Data columns (total 16 columns):
 #   Column                                 Non-Null Count  Dtype  
---  ------                                 --------------  -----  
 0   Year                                   84 non-null     int64  
 1   hemisphere                             84 non-null     object 
 2   Extent                                 84 non-null     float64
 3   AnomalyLand                            84 non-null     float64
 4   AnomalyOcean                           84 non-null     float64
 5   Fossil and Industrial                  84 non-null     float64
 6   Agriculture Waste and Forest Burining  84 non-null     float64
 7   biomass                                84 non-null     float64
 8   brown_coal                             84 non-null     float64
 9   coal_coke                              84 non-null     float64
 10  diesel_oil                             84 non-null     float64
 11  hard_coa